docs/source/python/filesystems.rst

```diff
@@ -200,10 +240,34 @@ For example::
    import pyarrow.dataset as ds
    ds.dataset("data/", filesystem=fs)
 
+::
```

Comment:
Might make sense to move the "Similarly for Azure Blob Storage" phrase from a comment to an actual phrase before the `::` here? I guess it would lead to an easier to follow flow of documentation

Reply:
That makes sense yes, I've pushed the change.